astropy/constants/codata2017.py

```diff
@@ -0,0 +1,117 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+"""
+Astronomical and physics constants in SI units.  See :mod:`astropy.constants`
+for a complete listing of constants defined in Astropy.
+"""
+import math
+
+from .constant import Constant, EMConstant
+
+
+# PHYSICAL CONSTANTS
+
+class CODATA2017(Constant):
+    default_reference = 'CODATA 2017'
+    _registry = {}
+    _has_incompatible_units = set()
+
+
+class EMCODATA2017(CODATA2017, EMConstant):
+    _registry = CODATA2017._registry
+
+
+# Table 3 from Newell, D. B., et al. 2018, Metrologia, 55, L13
+# http://iopscience.iop.org/article/10.1088/1681-7575/aa950a/meta
+
+h = CODATA2017('h', "Planck constant", 6.62607015e-34,
+               'J s', 0.0, system='si')
+
+e = EMCODATA2017('e', 'Electron charge', 1.602176634e-19,
+                 'C', 0.0, system='si')
+
+k_B = CODATA2017('k_B', "Boltzmann constant", 1.380649e-23,
+                 'J / (K)', 0.0, system='si')
+
+N_A = CODATA2017('N_A', "Avogadro's number", 6.02214076e23,
+                 '1 / (mol)', 0.0, system='si')
+
+# https://en.wikipedia.org/wiki/2019_redefinition_of_SI_base_units
+
+c = CODATA2017('c', "Speed of light in vacuum", 299792458.,
+               'm / (s)', 0.0, system='si')
+
+# Derived from above.
+
+hbar = CODATA2017('hbar', "Reduced Planck constant", h.value / (2 * math.pi),
+                  'J s', 0.0, system='si')
+
+sigma_sb = CODATA2017(
+    'sigma_sb', "Stefan-Boltzmann constant",
+    2 * math.pi ** 5 * k_B.value ** 4 / (15 * h.value ** 3 * c.value ** 2),
+    'W / (K4 m2)', 0.0, system='si')
+
+R = CODATA2017('R', "Gas constant", k_B.value * N_A.value,
+               'J / (K mol)', 0.0, system='si')
+
+# Same as CODATA2014.
```

Comment:
These are mainly the same as 2014 because I cannot find any clear indication that they have been changed. If you find otherwise, please correct me.